### PR TITLE
UI: Ember 3.0 -> 3.4

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
Officially on the latest LTS 🎉

The bulk of the work here was making tests pass since some things mostly in Ember Data changed. Additionally, I addressed the immediate deprecations, opened a [PR in ivy-codemirror](https://github.com/IvyApp/ivy-codemirror/pull/40), and ran the ES5 Getters codemod.

### CODE REVIEW

File: ui/tests/unit/adapters/job-test.js [hunk 1]
```diff
@@ -383,6 +383,10 @@ module('Unit | Adapter | Job', function(hooks) {
     const region = 'region-2';
     window.localStorage.nomadActiveRegion = region;
 
+    // Regions are fetched in the before hook, so manually dirty the activeRegion
+    // instead of repeating what is in the beforeEach here.
+    this.system.notifyPropertyChange('activeRegion');
```
Comment (alisdair) [inline ui/tests/unit/adapters/job-test.js]: This jumped out at me a bit, and I had to go and read the system service to understand what's going on here. Is there any alternative to `notifyPropertyChange` here? For example, might it make sense to call `this.system.set('activeregion', region)`, or are you trying to test that the computed property fetches the region from local storage?

Comment (DingoEatingFuzz) [reply ui/tests/unit/adapters/job-test.js]: Specifically trying to test that the computed property fetches the region from local storage. 

The alternative is to get namespaces after setting the localStorage property, but that means all my URL checks need to include namespaces, or omit the first entry.

This is not my favorite but it made for the smallest diff and least repetitive assertions. Maybe it's still worth refactoring.

**Comment by DingoEatingFuzz** [comment]
@alisdair, I decided to refactor those unit tests to avoid the `notifyPropertyChange` workaround.

**Comment by alisdair** [comment]
@DingoEatingFuzz Nice! More explicit and definitely clearer to me 😃

File: ui/config/deprecation-workflow.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ self.deprecationWorkflow.config = {
     { handler: 'throw', matchId: 'ember-inflector.globals' },
     { handler: 'throw', matchId: 'ember-runtime.deprecate-copy-copyable' },
     { handler: 'throw', matchId: 'ember-console.deprecate-logger' },
-    { handler: 'silence', matchId: 'ember-component.send-action' },
```
Comment (meirish) [inline ui/config/deprecation-workflow.js]: TIL about this file

Comment (DingoEatingFuzz) [reply ui/config/deprecation-workflow.js]: From this addon! https://github.com/mixonic/ember-cli-deprecation-workflow

Super helpful for triaging deprecations. I don't think you're supposed to commit it like this, but I'm gonna try it out. My main fear is deprecations related to `str.pluralize()` creeping back in. This makes it a runtime error now.

**Pull request merged**